src/Plugin/TypedDataFormWidget/TextareaWidget.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+namespace Drupal\typed_data\Plugin\TypedDataFormWidget;
+
+use Drupal\Core\Form\SubformStateInterface;
+use Drupal\Core\TypedData\DataDefinition;
+use Drupal\Core\TypedData\DataDefinitionInterface;
+use Drupal\Core\TypedData\Type\FloatInterface;
+use Drupal\Core\TypedData\Type\IntegerInterface;
+use Drupal\Core\TypedData\Type\StringInterface;
+use Drupal\Core\TypedData\TypedDataInterface;
+use Drupal\typed_data\Form\SubformState;
+use Drupal\typed_data\Widget\FormWidgetBase;
+use Symfony\Component\Validator\ConstraintViolationListInterface;
+
+/**
+ * @TypedDataFormWidget(
+ *   id = "textarea",
+ *   label = @Translation("Textarea"),
+ *   description = @Translation("A multi line text input widget."),
+ * )
+ */
+class TextareaWidget extends FormWidgetBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function defaultConfiguration() {
+    return parent::defaultConfiguration() + [
+      'label' => NULL,
+      'description' => NULL,
+      'placeholder' => NULL,
+      'rows' => 5,
+      'cols' => 60,
+      'resizable' => 'both',
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function isApplicable(DataDefinitionInterface $definition) {
+    return is_subclass_of($definition->getClass(), StringInterface::class) ||
+      is_subclass_of($definition->getClass(), IntegerInterface::class) ||
```

Comment:
I don't think it makes sense to edit floats or integers in a textarea. Let's support strings only.

Reply:
this still holds, so the code is wrong.